src/libsyntax/feature_gate.rs

```diff
@@ -108,124 +108,227 @@ macro_rules! declare_features {
 // was set. This is most important for knowing when a particular feature became
 // stable (active).
 //
+// Note that the features should be grouped into internal/user-facing
+// and then sorted by version inside those groups.
+// FIXME(60361): Enforce ^-- with tidy.
+//
 // N.B., `tools/tidy/src/features.rs` parses this information directly out of the
 // source, so take care when modifying it.
 
 declare_features! (
-    (active, asm, "1.0.0", Some(29722), None),
-    (active, concat_idents, "1.0.0", Some(29599), None),
-    (active, link_args, "1.0.0", Some(29596), None),
-    (active, log_syntax, "1.0.0", Some(29598), None),
-    (active, non_ascii_idents, "1.0.0", Some(55467), None),
-    (active, plugin_registrar, "1.0.0", Some(29597), None),
-    (active, thread_local, "1.0.0", Some(29594), None),
-    (active, trace_macros, "1.0.0", Some(29598), None),
+    // -------------------------------------------------------------------------
+    // Internal feature gates.
+    // -------------------------------------------------------------------------
 
-    // rustc internal, for now
+    // no tracking issue START
+
+    // Allows using the `rust-intrinsic`'s "ABI".
     (active, intrinsics, "1.0.0", None, None),
+
+    // Allows using `#[lang = ".."]` attribute for linking items to special compiler logic.
     (active, lang_items, "1.0.0", None, None),
-    (active, format_args_nl, "1.29.0", None, None),
 
-    (active, link_llvm_intrinsics, "1.0.0", Some(29602), None),
-    (active, linkage, "1.0.0", Some(29603), None),
+    // Allows using the `#[stable]` and `#[unstable]` attributes.
+    (active, staged_api, "1.0.0", None, None),
 
-    // rustc internal
+    // Allows using `#[allow_internal_unstable]`. This is an
+    // attribute on `macro_rules!` and can't use the attribute handling
+    // below (it has to be checked before expansion possibly makes
+    // macros disappear).
+    (active, allow_internal_unstable, "1.0.0", None, None),
+
+    // Allows using `#[allow_internal_unsafe]`. This is an
+    // attribute on `macro_rules!` and can't use the attribute handling
+    // below (it has to be checked before expansion possibly makes
+    // macros disappear).
+    (active, allow_internal_unsafe, "1.0.0", None, None),
+
+    // Allows using the macros:
+    // + `__diagnostic_used`
+    // + `__register_diagnostic`
+    // +`__build_diagnostic_array`
     (active, rustc_diagnostic_macros, "1.0.0", None, None),
+
+    // Allows using `#[rustc_const_unstable (feature)]` which lets
```

Comment:
```suggestion
    // Allows using `#[rustc_const_unstable(feature)]` which lets
```

Reply:
Tidy got angry with me when I actually wrote it this way which is why I changed it ;)

Reply:
?? why would tidy complain about this?

Reply:
I get: `tidy error: /home/centril/programming/rust/master/src/libsyntax/feature_gate.rs:152: malformed stability attribute` cause tidy looks for lines containing `rustc_const_unstable(`.

Reply:
Fixed it by using a more well formed example.